src/class.h

```diff
@@ -48,14 +48,17 @@ class Thumbnailer {
                       // byte budget.
   };
 
-  // Adds a frame with a timestamp (in millisecond).
+  // Adds a frame with a timestamp (in millisecond). The 'pic' argument must
+  // outlive the last GenerateAnimation() call.
   Status AddFrame(const WebPPicture& pic, int timestamp_ms);
 
   // Generates the animation.
   Status GenerateAnimationNoBudget(WebPData* const webp_data);
 
   // Finds the best quality that makes the animation fit right below the given
-  // byte budget and generates the animation.
+  // byte budget and generates the animation. The 'webp_data' argument is
+  // expected to be initialized (otherwise WebPDataClear() might free some
```

Comment:
The explanation inside the parenthesis is a bit too heavy for a method comment, it could have been omitted.